apps/web-client/src/lib/db/cr-sqlite/customers.ts

```diff
@@ -251,43 +253,73 @@ export const removeBooksFromCustomer = async (db: DB, customerId: number, bookId
 
 /**
  * Marks customer order lines as received when supplier order lines are fulfilled.
- * For each supplied ISBN, it updates the earliest unfulfilled customer order line
+ * For each supplied rowId, it updates the earliest unfulfilled customer order line
  * (that has been placed but not received) with the current timestamp as received date.
  *
  * @param {DB} db - The database connection instance
- * @param {isbns[]} isbns - Array of supplier order line isbns that have been received
+ * @param {rowIds[]} rowIds - Array of supplier order line ids that have been received
  * @returns {Promise<void>} A promise that resolves when all relevant customer orders are marked as received
  *
  * @remarks
  * - Only updates the earliest unfulfilled order line for each ISBN
  * - Only updates order lines that have been placed but not yet received
  * - Updates are performed in a single transaction
  * - If supplierOrderLines is empty, the function returns immediately
+ */
+
+export const markCustomerOrderAsReceived = async (db: DB, rowIds: number[]): Promise<void> => {
+	if (!rowIds.length) return;
+
+	return db.tx(async (txDb) => {
+		const timestamp = Date.now();
+		const placeholders = multiplyString("?", rowIds.length);
+		await txDb.exec(
+			`
+		 UPDATE customer_order_lines
+SET received = ?
+WHERE id IN (
+    SELECT id FROM customer_order_lines
+    WHERE id IN (${placeholders})
+        AND placed IS NOT NULL
+        AND received IS NULL
+    ORDER BY placed ASC
+    LIMIT 1
+);`,
+			[timestamp, ...rowIds]
+		);
+	});
+};
```

Comment:
2 points here: 
- can we use customer order line `id` here (name the param `ids` as well...`rowIds` is a bit opaque)
- when using `customer_order_lines.id` - it is unique, no need for the subquery, the sql could simply set received where `id` (in table) in `ids` (param):
```ts
const markAsReceived(db: DB, ids: string[]) => {
  // ...
  
  db.exec(`
      UPDATE ... SET ... WHERE id IN (${multiplyString("?", ids.length)})
    `,
    [timestamp, ...ids]
  )
}
```